### ISSUE TITLE
When loading a savegame, Player might skip a command in each interpreter

### ISSUE THREAD

**Issue by Zegeri** [EasyRPG/Player#757]
https://github.com/EasyRPG/Player/blob/126265e206b1/src/game_interpreter_map.cpp#L118 increments the current_command index of every top interpreter. This index is not decremented at loading, so it might skip a command like ProceedWithMovement (from now: PWM) or KeyInputProc (from now: KIP) and mess things up. The increment is done to match the savegames produced by RPG_RT.

There's a catch. In Player: while PWM or KIP are running, the interpreter index points to its line. While Wait is running, the index points to the next line. In our savegames, the current_command will point to the next line of PWM and KIP and the second next line of Wait.

In RPG_RT's savegames, the index points to the next line in all three cases. We can guess that RPG_RT isn't parsing the command each frame like we do, but it uses the local variables we identified as SaveEventData::unknown_0d_move_waiting, keyinput_*, time_left for PWM, waiting KIP and Wait respectively.

This must affect...a lot of games, but I can confirm it affects Monigote Fantasy.

**Comment by Ghabry** [comment]
I can confirm your guess because I noticed this when writing the event tracer (by stepping though the ASM code). RPG_RT never goes in the event code again but has if-checks for these flags before and exits earlier.

**Comment by Ghabry** [comment]
This is even worse. Besides skipping lines it also ignores all flow changes.

**Comment by Ghabry** [comment]
My fix only handles the really bad bug: skipping of an event command.

The save will now point at the "KeyInput" line. Which is wrong (when loading in RPG_RT). The proper fix here is not in the savegame code but altering KeyInputProc to go to the next line and handle it as zegeri suggested it.

**Comment by fmatthew5876** [comment]
Here are concrete test cases:

Create this parallel event:
```
Wait 0.0s
Key Input Processing, Shift Key, Wait
Text: Hello
```

Start a new game in RPG_RT, save your game from the menu.
Saved chunks include:

```
current_command=2
keyinput_wait=1
```

Load the game from `RPG_RT`, continues waiting.
Load the game from `Player`, "Hello" message is displayed.

**Comment by Ghabry** [comment]
You also see in the dynrpg event tracer that RPG_RT does e.g. the "wait for key input" processing outside of the event command.

**Comment by fmatthew5876** [comment]
Do my 3 test cases adequately cover this issue? Did I miss anything?

**Comment by fmatthew5876** [comment]
Fixed by #1740
